### PR TITLE
Place block storage (performance ISCSI) order by indicating iops, disksize, hourlybillingflag

### PR DESCRIPTION
**Pull request by jianqiu**
This PR is trying to fix the issue [110](https://github.com/maximilien/softlayer-go/issues/110)

### CODE REVIEW

File: softlayer/softlayer_product_package_service.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ type SoftLayer_Product_Package_Service interface {
 
 	GetItemPrices(packageId int) ([]datatypes.SoftLayer_Product_Item_Price, error)
 	GetItemPricesBySize(packageId int, size int) ([]datatypes.SoftLayer_Product_Item_Price, error)
+	GetItemPricesBySizeAndIops(packageId int, size int, capacity int) ([]datatypes.SoftLayer_Product_Item_Price, error)
```
Comment (maximilien) [inline softlayer/softlayer_product_package_service.go]: Hi, @jianqiu, are these service methods in the `SL_Product_Item_Price` service? If yes, then this PR makes sense and should be merged in SL go, if not then let's move these into the SL CPI code. Maybe create them into a file in the `utils` package.  Thoughts?

File: test_helpers/mock_softlayer_product_package.go [hunk 2]
```diff
@@ -29,6 +29,14 @@ func (mock *MockProductPackageService) GetItemPricesBySize(packageId int, size i
 	return []datatypes.SoftLayer_Product_Item_Price{}, errors.New("Not supported")
 }
 
+func (mock *MockProductPackageService) GetItemPricesBySizeAndIops(packageId int, size int, capacity int) ([]datatypes.SoftLayer_Product_Item_Price, error) {
+	return []datatypes.SoftLayer_Product_Item_Price{}, errors.New("Not supported")
+}
+
+func (mock *MockProductPackageService) GetIopsItemPricesBySize(packageId int, size int) ([]datatypes.SoftLayer_Product_Item_Price, error) {
```
Comment (maximilien) [inline test_helpers/mock_softlayer_product_package.go]: Let's use Fake as we do for all other mocks/fakes. For consistency.

**Comment by maximilien** [comment]
Overall looks good @jianqiu, however, see my comments. I think this needs to be in CPI as utility unless the answer to my question is yes and you can point me to SL docs online. Thx

File: softlayer/softlayer_product_package_service.go (hunk 1 above)
Comment (jianqiu) [inline softlayer/softlayer_product_package_service.go]: @maximilien ,
I think it should be a complex service method in the `SoftLayer_Product_Package`. The based method is [getItemPrices](http://sldn.softlayer.com/reference/services/SoftLayer_Product_Package/getItemPrices), what we do is to customize the method with result filter and mask. The following SL blog is the example we followed.
https://softlayer.github.io/python/place_order_block_storage_performance_iscsi/

**Comment by maximilien** [comment]
Ok @jianqiu what about this proposed compromise:

1. We keep it here as you want since, as you argue, it's an advance use of the SL API
2. We, however, move all non-official SL APIs functions into a new `utils` package

That way we do not break the main design point of SL go to reflect the SL APIs 1-to-1 as much as possible, yet allow useful "extensions" like this one to exist into utilities. We could even in time create things like `utils/storage` and `utils/virtual_guest` etc.

What do you think JQ?

**Comment by jianqiu** [comment]
@maximilien ，
Max, I already updated the PR and removed all *special* methods, made just 1-to-1 mapping. And the PR had resolved issue [110](https://github.com/maximilien/softlayer-go/issues/110), Simyung Yang already closed it. Thanks!

File: services/fakes/fake_softlayer_product_package.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+func (fps *FakeProductPackageService) GetItemsByType(packageType string) ([]datatypes.SoftLayer_Product_Item, error) {
+	response, _ := testhelpers.ReadJsonTestFixtures("services", "SoftLayer_Product_Package_getItemsByType_virtual_server.json")
+
+	productItems := []datatypes.SoftLayer_Product_Item{}
+	json.Unmarshal(response, &productItems)
+
+	return productItems, nil
+}
+
+func (fps *FakeProductPackageService) GetItemPrices(packageId int, filters string) ([]datatypes.SoftLayer_Product_Item_Price, error) {
```
Comment (maximilien) [inline services/fakes/fake_softlayer_product_package.go]: Why are we not supporting these methods? Just don't include them in the interface, that's better than returning an error IMO

**Comment by maximilien** [comment]
Merging for purposes of solving issue #110 but please address my comment, remove the methods we are not implementing rather than returning some special errors.

**Pull request merged**